clean_J2 branch move function version 확인 요망

wlals7217 opened the pull request:
v0.1.03 move function version 변경사항

- 블록 이동 및 병합 기능을 하는 부분을 move 함수로 정의, main.c의 코드 대체
- move 함수 부속함수 정의
- 보드 저장, 출력, 스코어 계산부분 임시 선언 또는 정의
- 가독성 향상을 위한 연산자 좌우 띄어쓰기 정리

길고 중복코드가 많은 방향키 기능을 일반화하기 위해 네 방향키의 내부기능들을  함수를 함수화해서 일반화합니다. main.c에 적용된 것은 아직 move함수 하나입니다.
방향키 'd'는 아직 버그가 있습니다.
피드백 부탁드려요!
@1000peach @LeeEunhyung

1000peach committed:
Make function (setUp, allocateArr)

- setUp, allocateArr 함수로 동적할당 및 랜덤 수 생성하는 초기화 부분을 분리
- 현재까지 commit 내용의 설명을 주석으로 추가

wlals7217 committed:
make function (move, push, merge, printBoard, save, updateScore)

-move 함수 정의.
-move 부속함수push, merge, printboard, save, updateScore 선언.

wlals7217 committed:
make function (move, push, merge, checkMove)

-move함수내부의 스코어계산관련함수 제거.
-push 함수, merge함수 메인에서 분리.
-move함수 벡터쓰면 과하게 복잡한 함수가 될 것. 부속함수에서 분기 하도록
수정.
-연산자 좌우 띄어쓰기 수정.

LeeEunhyung commented:
clean_J2 확인했습니다.
코드를 봤는데 merge()와 push() 함수에서 네 방향을 한꺼번에 함수화 하면 
key 데이터 확인 switch문을 inputKey() 함수에서 한 번, 위 함수들에서 한 번, 
총 두 번 하게 되는 것같아서요!
merge() 함수랑 push() 함수를 각각 네 방향으로 나누는 건 어떨까요?!
제가 inputKey() 함수 만들어서 pull request하겠습니다!
또 InitGame.c 파일에서 printBoard() 함수가 파라미터값이 다른 상태로 두 개가 정의되어 있던데 그 부분도 수정해주시면 좋을 것같아요!
수고하셨습니다!!~

1000peach commented:
확인했습니다!
저는 한 줄 주석은 /**/보다 //로 작성하고, 초반에 정의했던 규칙 대로 중괄호 닫는 부분에 주석 위치를 변경하면 좋을 것 같아요.
개선 내용이 많은데 고생하셨습니다.

wlals7217 commented:
LeeEunhyung님 일반화 때문에 분기는 불가피 하지만 함수를 네 방향으로 쪼개는 것은 장단점이 있을 것 같습니다. 우선은 가독성과 캡슐화라는 장점을 봐주시면 좋을 것 같고 printBoard 함수는 고치겠습니다.
1000peach님 주석도 고치겠습니다!

wlals7217 committed:
Clean code

-printBoard 함수의 중복을 제거하고, InitGame.h에 선언합니다.
-한 줄 짜리주석을 '//'로 시작하도록 바꾸고 위치를 맞춥니다.
-잘못된 pch.h include 제거
